packages/ui-components/src/components/UIList/UIList.tsx

```diff
@@ -21,13 +21,13 @@ export {
     IGroupRenderProps as UIGroupRenderProps
 };
 
-export interface ListProps extends IGroupedListProps {
+export type ListProps = Omit<IGroupedListProps, 'onRenderCell'> & {
     groups: IGroup[];
     groupProps?: IGroupRenderProps;
     items: never[];
     onSelect: (value: IGroup) => void;
     onRenderCell: (nestingDepth?: number, item?: never, index?: number) => React.ReactNode;
```

Comment:
Could this be simplified? Not sure if that extensive redefinition of `IGroupedListProps` is still needed 
```suggestion
export type ListProps = IGroupedListProps & {
    items: never[];
    onSelect: (value: IGroup) => void;
};
```

Reply:
At least the `onRenderCell` type of the property `item` is changed to `never`. If this is needed? 🤷🏻

Reply:
You're right, overlooked the `never` on the item property. Then only `groups` and `groupProps` definition were already included in `IGroupedListProps`

Reply:
So here are the differences:

1. property `groups` is optional in `IGroupedListProps` but mandatory in `ListProps` 🚫 
2. property `groupProps?` is identical and could be deleted ✅ 
3. property `items` is `any` in `IGroupedListProps` but `never` in `ListProps` 🚫
4. function `onSelect` is not part of `IGroupedListProps` and added in `ListProps` 🚫
5. function `onRenderCell` has optional property `item` as `any` in `IGroupedListProps` but `never` in `ListProps` 🚫

So only 2. could be deleted but to be honest I only want to do the bare minimum refactoring in this PR to get the adjusted TS setup running. Let's avoid extending the scope to a more general refactoring to not further increase complexity.